cli: better server defaults

Currently we default to `reana.cern.ch` as the REANA deployment URL:

```console
$ git grep reana.cern.ch reana_client
reana_client/cli/__init__.py:        server_url = os.environ.get('REANA_SERVER_URL', 'http://reana.cern.ch')
reana_client/cli/cwl_runner.py:    server_url = os.environ.get('REANA_SERVER_URL', 'http://reana.cern.ch')
```

This is not safe and could lead to people submitting unwillingly to the CERN deployment.

Instead, `localhost` might be safer default... or simply print a message such as:

[ERROR] REANA client is not connected to any REANA cluster. Please set REANA_SERVER_URL environment variable to the remote REANA cluster you would like to connect to. For example: `export REANA_SERVER_URL=https://reana.cern.ch/`.

--- a/reana_client/cli/__init__.py
+++ b/reana_client/cli/__init__.py
@@ -42,7 +42,15 @@ class Config(object):
 
     def __init__(self):
         """Initialize config variables."""
-        server_url = os.environ.get('REANA_SERVER_URL', 'http://reana.cern.ch')
+        server_url = os.environ.get('REANA_SERVER_URL', None)
+
+        if not server_url:
+            logging.error(
+                'REANA client is not connected to any REANA cluster.\n'
+                'Please set REANA_SERVER_URL environment variable to '
+                'the remote REANA cluster you would like to connect to.\n'
+                'For example: export REANA_SERVER_URL=https://reana.cern.ch/')
+            sys.exit(1)
 
         logging.info('REANA Server URL ($REANA_SERVER_URL) is: {}'
                      .format(server_url))

